backend/pkg/connect/get_connectors.go

```diff
@@ -221,33 +251,177 @@ func listConnectorsExpandedToClusterConnectorInfo(l map[string]con.ListConnector
-				TaskID:   task.ID,
-				State:    task.State,
-				WorkerID: task.WorkerID,
-				Trace:    task.Trace,
-			}
-			if task.State == connectorStateRunning {
-				runningTasks++
-			}
+		c := c
+		cInfo := connectorsResponseToClusterConnectorInfo(&c)
+		connectorInfo = append(connectorInfo, *cInfo)
+	}
+
+	return connectorInfo
+}
+
+//nolint:gocognit,cyclop // lots of inspection of state and tasks to determine status and errors
+func connectorsResponseToClusterConnectorInfo(c *con.ListConnectorsResponseExpanded) *ClusterConnectorInfo {
+	totalTasks := len(c.Status.Tasks)
+	tasks := make([]ClusterConnectorTaskInfo, totalTasks)
+	connectorTaskErrors := make([]ClusterConnectorInfoError, 0, totalTasks)
+
+	runningTasks := 0
+	failedTasks := 0
+	pausedTasks := 0
+	restartingTasks := 0
+	for i, task := range c.Status.Tasks {
+		tasks[i] = ClusterConnectorTaskInfo{
+			TaskID:   task.ID,
+			State:    task.State,
+			WorkerID: task.WorkerID,
+			Trace:    task.Trace,
+		}
+
+		switch task.State {
+		case connectorStateRunning:
+			runningTasks++
+		case connectorStateFailed:
+			failedTasks++
+
+			errTitle := fmt.Sprintf("Connector %s Task %d is in failed state.", c.Info.Name, task.ID)
+			connectorTaskErrors = append(connectorTaskErrors, ClusterConnectorInfoError{
+				Type:    connectorErrorTypeError,
+				Title:   errTitle,
+				Content: traceToErrorContent(errTitle, task.Trace),
+			})
+		case connectorStatePaused:
+			pausedTasks++
+		case connectorStateRestarting:
+			restartingTasks++
+		}
+	}
+
+	// LOGIC:
+	// HEALTHY: Connector is in running state, > 0 tasks, all of them in running state.
+	// UNHEALTHY: Connector is failed state.
+	//			Or Connector is in running state but has 0 tasks.
+	// 			Or Connector is in running state, has > 0 tasks, and all tasks are in failed state.
+	// DEGRADED: Connector is in running state, has > 0 tasks, but has at least one state in failed state, but not all tasks are failed.
+	// PAUSED: Connector is in paused state, and all tasks are in paused state.
+	// RESTARTING: Connector is in restarting state, or at least one task is in restarting state.
+	var connStatus connectorStatus
+	var errDetailedContent string
+	//nolint:gocritic // this if else is easier to read as they map to rules and logic specified above.
+	if (c.Status.Connector.State == connectorStateRunning) &&
+		totalTasks > 0 && runningTasks == totalTasks {
+		connStatus = connectorStatusHealthy
+	} else if (c.Status.Connector.State == connectorStateFailed) ||
+		((c.Status.Connector.State == connectorStateRunning) && (totalTasks == 0 || totalTasks == failedTasks)) {
+		connStatus = connectorStatusUnhealthy
+		if totalTasks == 0 {
+			errDetailedContent = "Connector is in " + strings.ToLower(c.Status.Connector.State) + " state but has no tasks."
+		} else if totalTasks == failedTasks {
+			errDetailedContent = "Connector is in " + strings.ToLower(c.Status.Connector.State) + " state but all tasks are in failed state."
 		}
```

Comment:
I wonder whether `else if (c.Status.Connector.State == connectorStateFailed)` should be its own case. If the connector state is in failed state we know that our own state is going to be failed too right? It would also read weirdly if we have an error saying: "Connector is in failed state but has no tasks running" no?